workspaces/ballerina/type-editor/src/TypeEditor/FieldEditor.tsx

```diff
@@ -183,8 +183,21 @@ export const FieldEditor: React.FC<FieldEditorProps> = (props) => {
                 <CollapsibleSection>
                     <TextField label='Default Value' value={member.defaultValue} onChange={handleMemberDefaultValueChange} style={{ width: '180px' }} />
                     <TextField label='Description' value={member.docs} onChange={handleDescriptionChange} style={{ width: '180px' }} />
+                    <CheckBox
+                        sx={{ border: 'none', padding: '5px' }}
+                        label="Readonly"
+                        checked={member.readonly}
+                        onChange={(checked: boolean) => {
+                            // Match the same pattern used in the working checkbox
+                            onChange({
+                                ...member,
+                                readonly: checked 
+                                }
+                            );
+                        }}
+                    />
                 </CollapsibleSection>
-            )}
+            )}    
```

Comment:
Unnecessary white spaces?

```suggestion
            )}
```